csharp/src/Microsoft.ML.OnnxRuntime/NamedOnnxValue.cs

```diff
@@ -218,12 +218,22 @@ out nativeTensor
                                                     ));
 
                     // fill the native tensor, using GetValue(index) from the Tensor<string>
-                    string[] stringsInTensor = new string[tensorValue.Length];
+                    var len = tensorValue.Length;
+                    var stringsInTensor = new IntPtr[len];
+                    var pinnedHandles = new GCHandle[len];
                     for (int i = 0; i < tensorValue.Length; i++)
                     {
-                        stringsInTensor[i] = tensorValue.GetValue(i);
+                        var utf8str = UTF8Encoding.UTF8.GetBytes(tensorValue.GetValue(i) + "\0");
```

Comment:
>"\0" [](start = 91, length = 4)

This is a a double byte zero, can we add a single byte?

Reply:
The zero has to be in UTF-16 to start off with, before can call GetBytes(), so double byte seems OK.